contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -95,6 +121,36 @@ procedure Lkt_Toolbox is
       end if;
    end Print_Semantic_Result;
 
+   -------------------------------
+   -- Populate_Invalid_Decl_Map --
+   -------------------------------
+
+   function Populate_Invalid_Decl_Map
+      (Node : Analysis.LK_Node'Class) return Common.Visit_Status is
+      use type Common.LK_Node_Kind_Type;
+   begin
+      --  Populate ``Invalid_Decl_Map`` with declarations annotated with
+      --  ``@invalid``.
+
+      if Node.Kind = Common.Lkt_Full_Decl
+         and then Node.As_Full_Decl.P_Has_Annotation
+            (To_Unbounded_Text ("invalid"))
+      then
+         --  ``P_Topmost_Invalid_Decl`` should return the same node. In that
+         --  case, include this node in the map, otherwise nested ``@invalid``
+         --  declaration has been detected: raise an error.
+
+         if Invalid_Decl_Map.Contains
+               (Node.P_Topmost_Invalid_Decl.As_LK_Node)
```

Comment:
The other map operation above does not need the `As_Lk_Node` conversion: why is it needed here? (same below)

Reply:
As discussed live the other day, you can modify `Node`'s declaration to be a simple `LK_Node`, and initialize it with `Analysis.Node (S).As_LK_Node`.

Reply:
I still end up with a compilation error when trying to get rid of this `As_LK_Node` conversion [here](https://github.com/AdaCore/langkit/blob/f87bb05f218fc176d8986d872e72c2238886f164/contrib/lkt/extensions/mains/lkt_toolbox.adb#L177). As it is used in the traversal function, `Node` should be of type `LK_Node'Class` no? I guess we should discuss it again.

Reply:
That’s correct: Node being a `LK_Node'Class` value because of the `Traverse` subprogram (and you can’t change that), you need to call `.As_LK_Node` to pass it as a `LK_Node` value.

It seems that GitHub won’t let me see the original commit for which I wrote “same below”, so I can’t double check: sorry if I was misleading. The simplification was valid for `Node.P_Topmost_Invalid_Decl` in any case.